### ISSUE TITLE
can copy .txt but not .exe file from mount.cifs mounted virtio-fs drive

### ISSUE THREAD

**Issue by Dave--G** [virtio-win/kvm-guest-drivers-windows#582]
I'm mounting a cifs(smb3) file share into a zfs directory that's shared into a windows guest using virtio-fs.
It appears to work.  I can copy a text file into and out of the shared directory.

But when I try to copy an .exe or .msi file from the smb host via the folder, windows generates a "Cant read from source file or disk" error.

However if I copy a .exe file from the guest to the smb host via the folder, I can then copy it back out again.  When I look at the security properties for both files (via rdp into the windows smb server) they are identical.

Is this Windows UAC or antivirus somehow preventing the copy?  I tried turning both off and it didn't make any difference.

I'm running virtiofs.exe from a batch file I run after logging in to the windows guest to work around the read-only problem that happens when it's run as a service.  Could this me part of the problem?

My smb share is mounted in /etc/fstab like this (/mnt/data/share the the vm.xml `<filesystem>` source path:
//192.168.1.200/users /mnt/data/share/users cifs credentials=/etc/cifs-credentials,file_mode=0777,file_mode=0777,dir_mode=0777,vers=3.1.1,rw,noperm,iocharset=utf8,soft,use 0 0

Any suggestions for resolving this issue?

**Comment by Dave--G** [comment]
Log from point of copy prog.exe initiated to error message displayed:
*** VirtFsFuseRequest: >>req: 44 unique: 1595 len: 80
*** VirtFsFuseRequest: <<len: 16 error: 0 unique: 1595
*** GetSecurityByName: "\share\users\test\prog.exe"
*** VirtFsFuseRequest: >>req: 1 unique: 1596 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1596
*** SubmitLookupRequest: nodeid=65 ino=40760 size=6 blocks=50 atime=1621461270 mtime=1621458660 ctime=1621458660 atimensec=896310976 mtimensec=634507636 ctimensec=634507636 mode=41fd nlink=6 uid=1000 gid=1000 rdev=0 blksize=16384
*** VirtFsFuseRequest: >>req: 1 unique: 1597 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1597
*** SubmitLookupRequest: nodeid=146 ino=562949953421392 size=0 blocks=0 atime=1621065589 mtime=1621065589 ctime=1621065589 atimensec=451606300 mtimensec=451606300 ctimensec=451606300 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1598 len: 45
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1598
*** SubmitLookupRequest: nodeid=50 ino=281474976710747 size=0 blocks=0 atime=1621482326 mtime=1621482326 ctime=1621482326 atimensec=353733600 mtimensec=353733600 ctimensec=353733600 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1599 len: 70
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1599
*** SubmitLookupRequest: nodeid=152 ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** Open: "\share\users\test\prog.exe" CreateOptions: 0x01000064 GrantedAccess: 0x00120089
*** VirtFsFuseRequest: >>req: 1 unique: 1600 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1600
*** SubmitLookupRequest: nodeid=65 ino=40760 size=6 blocks=50 atime=1621461270 mtime=1621458660 ctime=1621458660 atimensec=896310976 mtimensec=634507636 ctimensec=634507636 mode=41fd nlink=6 uid=1000 gid=1000 rdev=0 blksize=16384
*** VirtFsFuseRequest: >>req: 1 unique: 1601 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1601
*** SubmitLookupRequest: nodeid=146 ino=562949953421392 size=0 blocks=0 atime=1621065589 mtime=1621065589 ctime=1621065589 atimensec=451606300 mtimensec=451606300 ctimensec=451606300 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1602 len: 45
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1602
*** SubmitLookupRequest: nodeid=50 ino=281474976710747 size=0 blocks=0 atime=1621482326 mtime=1621482326 ctime=1621482326 atimensec=353733600 mtimensec=353733600 ctimensec=353733600 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1603 len: 70
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1603
*** SubmitLookupRequest: nodeid=152 ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 14 unique: 1604 len: 48
*** VirtFsFuseRequest: <<len: 32 error: 0 unique: 1604
*** Open: fh: 2 nodeid: 152
*** SetFileInfo: ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** GetSecurity: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 3 unique: 1605 len: 56
*** VirtFsFuseRequest: <<len: 120 error: 0 unique: 1605
*** Read: Offset: 0 Length: 32768
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1606 len: 32808
*** VirtFsFuseRequest: <<len: 32784 error: 0 unique: 1606
*** Read: BytesTransferred: 32768
*** Read: Offset: 8069120 Length: 32768
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1607 len: 32808
*** VirtFsFuseRequest: <<len: 28707 error: 0 unique: 1607
*** VirtFsFuseRequest: Bytes Returned: 28707 Expected: 32784
*** Read: BytesTransferred: 28691
*** Read: Offset: 65536 Length: 32768
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1608 len: 32808
*** VirtFsFuseRequest: <<len: 32784 error: 0 unique: 1608
*** Read: BytesTransferred: 32768
*** Read: Offset: 32768 Length: 32768
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1609 len: 32808
*** VirtFsFuseRequest: <<len: 32784 error: 0 unique: 1609
*** Read: BytesTransferred: 32768
*** Read: Offset: 98304 Length: 32768
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1610 len: 32808
*** VirtFsFuseRequest: <<len: 32784 error: 0 unique: 1610
*** Read: BytesTransferred: 32768
*** Read: Offset: 131072 Length: 32768
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1611 len: 32808
*** VirtFsFuseRequest: <<len: 32784 error: 0 unique: 1611
*** Read: BytesTransferred: 32768
*** Read: Offset: 163840 Length: 65536
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1612 len: 65576
*** VirtFsFuseRequest: <<len: 65552 error: 0 unique: 1612
*** Read: BytesTransferred: 65536
*** Read: Offset: 229376 Length: 131072
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1613 len: 131112
*** VirtFsFuseRequest: <<len: 131088 error: 0 unique: 1613
*** Read: BytesTransferred: 131072
*** Read: Offset: 360448 Length: 262144
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1614 len: 262184
*** VirtFsFuseRequest: <<len: 262160 error: 0 unique: 1614
*** Read: BytesTransferred: 262144
*** Read: Offset: 622592 Length: 524288
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1615 len: 524328
*** Read: Offset: 0 Length: 512
*** Read: fh: 2 nodeid: 152
*** VirtFsFuseRequest: >>req: 15 unique: 1616 len: 552
*** VirtFsFuseRequest: <<len: 528 error: 0 unique: 1616
*** Read: BytesTransferred: 512
*** GetSecurityByName: "\share\users\test\prog.exe"
*** VirtFsFuseRequest: >>req: 1 unique: 1617 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1617
*** SubmitLookupRequest: nodeid=65 ino=40760 size=6 blocks=50 atime=1621461270 mtime=1621458660 ctime=1621458660 atimensec=896310976 mtimensec=634507636 ctimensec=634507636 mode=41fd nlink=6 uid=1000 gid=1000 rdev=0 blksize=16384
*** VirtFsFuseRequest: >>req: 1 unique: 1618 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1618
*** SubmitLookupRequest: nodeid=146 ino=562949953421392 size=0 blocks=0 atime=1621065589 mtime=1621065589 ctime=1621065589 atimensec=451606300 mtimensec=451606300 ctimensec=451606300 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1619 len: 45
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1619
*** SubmitLookupRequest: nodeid=50 ino=281474976710747 size=0 blocks=0 atime=1621482326 mtime=1621482326 ctime=1621482326 atimensec=353733600 mtimensec=353733600 ctimensec=353733600 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1620 len: 70
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1620
*** SubmitLookupRequest: nodeid=152 ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** Open: "\share\users\test\prog.exe" CreateOptions: 0x01200044 GrantedAccess: 0x00120089
*** VirtFsFuseRequest: >>req: 1 unique: 1621 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1621
*** SubmitLookupRequest: nodeid=65 ino=40760 size=6 blocks=50 atime=1621461270 mtime=1621458660 ctime=1621458660 atimensec=896310976 mtimensec=634507636 ctimensec=634507636 mode=41fd nlink=6 uid=1000 gid=1000 rdev=0 blksize=16384
*** VirtFsFuseRequest: >>req: 1 unique: 1622 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1622
*** SubmitLookupRequest: nodeid=146 ino=562949953421392 size=0 blocks=0 atime=1621065589 mtime=1621065589 ctime=1621065589 atimensec=451606300 mtimensec=451606300 ctimensec=451606300 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1623 len: 45
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1623
*** SubmitLookupRequest: nodeid=50 ino=281474976710747 size=0 blocks=0 atime=1621482326 mtime=1621482326 ctime=1621482326 atimensec=353733600 mtimensec=353733600 ctimensec=353733600 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1624 len: 70
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1624
*** SubmitLookupRequest: nodeid=152 ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 14 unique: 1625 len: 48
*** VirtFsFuseRequest: <<len: 32 error: 0 unique: 1625
*** Open: fh: 3 nodeid: 152
*** SetFileInfo: ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** Close: fh: 3 nodeid: 152
*** GetSecurityByName: "\share\users\test\prog.exe"
*** VirtFsFuseRequest: >>req: 18 unique: 1626 len: 64
*** VirtFsFuseRequest: >>req: 1 unique: 1627 len: 46
*** VirtFsFuseRequest: <<len: 16 error: 0 unique: 1626
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1627
*** SubmitLookupRequest: nodeid=65 ino=40760 size=6 blocks=50 atime=1621461270 mtime=1621458660 ctime=1621458660 atimensec=896310976 mtimensec=634507636 ctimensec=634507636 mode=41fd nlink=6 uid=1000 gid=1000 rdev=0 blksize=16384
*** VirtFsFuseRequest: >>req: 1 unique: 1628 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1628
*** SubmitLookupRequest: nodeid=146 ino=562949953421392 size=0 blocks=0 atime=1621065589 mtime=1621065589 ctime=1621065589 atimensec=451606300 mtimensec=451606300 ctimensec=451606300 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1629 len: 45
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1629
*** SubmitLookupRequest: nodeid=50 ino=281474976710747 size=0 blocks=0 atime=1621482326 mtime=1621482326 ctime=1621482326 atimensec=353733600 mtimensec=353733600 ctimensec=353733600 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1630 len: 70
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1630
*** SubmitLookupRequest: nodeid=152 ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** Open: "\share\users\test\prog.exe" CreateOptions: 0x01200064 GrantedAccess: 0x00120089
*** VirtFsFuseRequest: >>req: 1 unique: 1631 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1631
*** SubmitLookupRequest: nodeid=65 ino=40760 size=6 blocks=50 atime=1621461270 mtime=1621458660 ctime=1621458660 atimensec=896310976 mtimensec=634507636 ctimensec=634507636 mode=41fd nlink=6 uid=1000 gid=1000 rdev=0 blksize=16384
*** VirtFsFuseRequest: >>req: 1 unique: 1632 len: 46
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1632
*** SubmitLookupRequest: nodeid=146 ino=562949953421392 size=0 blocks=0 atime=1621065589 mtime=1621065589 ctime=1621065589 atimensec=451606300 mtimensec=451606300 ctimensec=451606300 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1633 len: 45
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1633
*** SubmitLookupRequest: nodeid=50 ino=281474976710747 size=0 blocks=0 atime=1621482326 mtime=1621482326 ctime=1621482326 atimensec=353733600 mtimensec=353733600 ctimensec=353733600 mode=41ff nlink=2 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 1 unique: 1634 len: 70
*** VirtFsFuseRequest: <<len: 144 error: 0 unique: 1634
*** SubmitLookupRequest: nodeid=152 ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** VirtFsFuseRequest: >>req: 14 unique: 1635 len: 48
*** VirtFsFuseRequest: <<len: 32 error: 0 unique: 1635
*** Open: fh: 3 nodeid: 152
*** SetFileInfo: ino=10414574138487089 size=8097811 blocks=0 atime=1570820976 mtime=1570820904 ctime=1570820904 atimensec=269797700 mtimensec=84630200 ctimensec=84630200 mode=81ff nlink=1 uid=0 gid=0 rdev=0 blksize=1048576
*** Close: fh: 3 nodeid: 152
*** VirtFsFuseRequest: >>req: 18 unique: 1636 len: 64
*** Close: fh: 2 nodeid: 152
*** Close: fh: 68 nodeid: 50
*** VirtFsFuseRequest: <<len: 16 error: 0 unique: 1636
*** VirtFsFuseRequest: >>req: 18 unique: 1637 len: 64
*** VirtFsFuseRequest: >>req: 29 unique: 1638 len: 64
*** VirtFsFuseRequest: <<len: 16 error: 0 unique: 1637
*** VirtFsFuseRequest: <<len: 16 error: 0 unique: 1638

**Comment by Dave--G** [comment]
(just guessing but should uid=0 gid=0 towards the later portion of the log?

**Comment by Dave--G** [comment]
This appears to be a wider problem, unrelated to cifs shares.  Leaving bug open but it can be closed if deemed not of value.

**Comment by MostHated** [comment]
I am having this same issue. Host has a cifs share mapped to my nas at /mnt/nasdata/ then I was passing that through to the guest. In addition to that, though, every time I try to save a file in my IDE, seemingly while/as it is in process of saving, I get a message saying "file was changed on disk" and it asks me to reload the file. I thought perhaps it was IDE related, but so far all roads seem to point to the virtio setup, especially now that I am seeing others with similar goofyness, such as this thread. 

Is there anything that could be done to help with this by chance, or am I just stuck with it for the time being? Not being able to save most of the time, or copy files from the drive to the guest is killing me, but unfortunately it's the only way I have found to access the host resources while connected to a VPN that disables any other network/usb connections, modify host file, etc.
